src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-flags-modal/osd-flags-modal.component.ts

```diff
@@ -0,0 +1,139 @@
+      description: 'OSDs that were previously marked out will not be marked back in when they start'
+    },
+    noout: {
+      code: 'noout',
+      name: 'No Out',
+      value: false,
+      description: 'OSDs will not automatically be marked out after the configured interval'
+    },
+    noup: {
+      code: 'noup',
+      name: 'No Up',
+      value: false,
+      description: 'OSDs are not allowed to start'
+    },
+    nodown: {
+      code: 'nodown',
+      name: 'No Down',
+      value: false,
+      description:
+        'OSD failure reports are being ignored, such that the monitors will not mark OSDs down'
+    },
+    pause: {
+      code: 'pause',
+      name: 'Pause',
+      value: false,
+      description: 'Pauses reads and writes'
+    },
+    noscrub: {
+      code: 'noscrub',
+      name: 'No Scrub',
+      value: false,
+      description: 'Scrubbing is disabled'
+    },
+    'nodeep-scrub': {
+      code: 'nodeep-scrub',
+      name: 'No Deep Scrub',
+      value: false,
+      description: 'Deep Scrubbing is disabled'
+    },
+    nobackfill: {
+      code: 'nobackfill',
+      name: 'No Backfill',
+      value: false,
+      description: 'Backfilling of PGs is suspended'
+    },
+    norecover: {
+      code: 'norecover',
+      name: 'No Recover',
+      value: false,
+      description: 'Recovery of PGs is suspended'
+    },
+    sortbitwise: {
+      code: 'sortbitwise',
+      name: 'Bitwise Sort',
+      value: false,
+      description: 'Use bitwise sort',
+      disabled: true
+    },
+    purged_snapdirs: {
+      code: 'purged_snapdirs',
+      name: 'Purged Snapdirs',
+      value: false,
+      description: 'OSDs have converted snapsets',
+      disabled: true
+    },
+    recovery_deletes: {
+      code: 'recovery_deletes',
+      name: 'Recovery Deletes',
+      value: false,
+      description: 'Deletes performed during recovery instead of peering',
+      disabled: true
+    }
+  };
+  flags: any[];
+  unknownFlags: string[] = [];
+
+  constructor(
+    public bsModalRef: BsModalRef,
+    private osdService: OsdService,
+    private notificationService: NotificationService
+  ) {}
+
+  ngOnInit() {
+    this.osdService.getFlags().subscribe((res: string[]) => {
+      res.forEach((value) => {
+        if (this.allFlags[value]) {
+          this.allFlags[value].value = true;
+        } else {
+          this.unknownFlags.push(value);
+        }
+      });
+      this.flags = _.toArray(this.allFlags);
+    });
+  }
+
+  submitAction() {
+    const newFlags = this.flags
+      .filter((flag) => flag.value)
+      .map((flag) => flag.code)
+      .concat(this.unknownFlags);
+
+    this.osdService.updateFlags(newFlags).subscribe(
+      () => {
+        this.notificationService.show(
+          NotificationType.success,
+          'OSD Flags were updated successfully.',
+          'OSD Flags'
+        );
+        this.bsModalRef.hide();
+      },
+      () => {
+        this.bsModalRef.hide();
```

Comment:
You should set a specific error description if it fails. So that the user knows something went wrong with the OSD settings. The generic message does not give any info that OSD flags couldn't be set.

Reply:
This will be done in a different PR.